plugin/storage/kafka/options_test.go

```diff
@@ -45,4 +53,92 @@ func TestFlagDefaults(t *testing.T) {
 	assert.Equal(t, defaultEncoding, opts.encoding)
+	assert.Equal(t, sarama.WaitForLocal, opts.config.RequiredAcks)
+	assert.Equal(t, sarama.CompressionNone, opts.config.Compression)
+	assert.Equal(t, 0, opts.config.CompressionLevel)
+}
+
+func TestCompressionLevelDefaults(t *testing.T) {
+	compressionLevel, err := getCompressionLevel("none", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["none"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("gzip", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["gzip"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("snappy", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["snappy"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("lz4", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["lz4"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("zstd", defaultCompressionLevel)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["zstd"].defaultCompressionLevel, compressionLevel)
+}
+
+func TestCompressionLevel(t *testing.T) {
+	compressionLevel, err := getCompressionLevel("none", 0)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["none"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("gzip", 4)
+	require.NoError(t, err)
+	assert.Equal(t, 4, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("snappy", 0)
+	require.NoError(t, err)
+	assert.Equal(t, compressionModes["snappy"].defaultCompressionLevel, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("lz4", 10)
+	require.NoError(t, err)
+	assert.Equal(t, 10, compressionLevel)
+
+	compressionLevel, err = getCompressionLevel("zstd", 20)
+	require.NoError(t, err)
+	assert.Equal(t, 20, compressionLevel)
+}
+
+func TestFailedCompressionLevelScenario(t *testing.T) {
+	_, err := getCompressionLevel("gzip", 14)
+	assert.Error(t, err)
+
+	_, err = getCompressionLevel("lz4", 18)
+	assert.Error(t, err)
+
+	_, err = getCompressionLevel("zstd", 25)
+	assert.Error(t, err)
+}
+
+func TestCompressionModes(t *testing.T) {
+	compressionModes, err := getCompressionMode("gzip")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.CompressionGZIP, compressionModes)
+
+	compressionModes, err = getCompressionMode("snappy")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.CompressionSnappy, compressionModes)
+
+	compressionModes, err = getCompressionMode("none")
+	require.NoError(t, err)
+	assert.Equal(t, sarama.CompressionNone, compressionModes)
+}
+
+func TestCompressionModeFailures(t *testing.T) {
+	_, err := getCompressionMode("test")
+	assert.Error(t, err)
+
+	_, err = getCompressionMode("temp")
+	assert.Error(t, err)
+}
+
+func TestRequiredAcksFailures(t *testing.T) {
```

Comment:
What about testing the valid values aswell?

Also same comment as above related to `test` and `temp`?

Reply:
fixed it. Also added test for valid values.